File: bot/providers/misc.py
from abc import abstractmethod


class Misc(object):
    @abstractmethod
    def is_process_running(self):
        raise NotImplementedError("is_process_running not implemented")

    @abstractmethod
    def start_process(self):
        raise NotImplementedError("start_process not implemented")

    @abstractmethod
    def kill_process(self):
        raise NotImplementedError("kill_process not implemented")

    @abstractmethod
    def ensure_resolutions_matches(self, img):
        raise NotImplementedError("ensure_resolutions_matches not implemented")

File: bot/providers/misc.go
package providers

import "gocv.io/x/gocv"

// Miscellaneous all functions that don't fall under actions or are bot specific
// fall under here
type Miscellaneous interface {
	IsProcessRunning() bool
	StartProcess()
	KillProcess()
	EnsureResolutionMatches(mat *gocv.Mat)
	// Ensures that every is ok
	PreCheck() error
}

type BaseMiscellaneous struct {
	provider Provider
}

func (*BaseMiscellaneous) PreCheck() error {
	panic("implement me")
}

func (*BaseMiscellaneous) IsProcessRunning() bool {
	//log.Panic("implement me")
	return true
}

func (*BaseMiscellaneous) StartProcess() {
	log.Panic("implement me")
}

func (*BaseMiscellaneous) KillProcess() {
	log.Panic("implement me")
}

func (*BaseMiscellaneous) EnsureResolutionMatches(mat *gocv.Mat) {
	log.Panic("implement me")
}

func NewMisc(o *Options) Miscellaneous {
	misc := new(DuelLinksMisc)
	misc.provider = o.Provider
	return misc
}
